common/src/main/java/io/netty5/util/concurrent/DefaultTaskScheduler.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty5.util.concurrent;
+
+import io.netty5.util.concurrent.AbstractScheduledEventExecutor.RunnableScheduledFutureNode;
+import io.netty5.util.internal.DefaultPriorityQueue;
+import io.netty5.util.internal.PriorityQueue;
+
+import java.util.Comparator;
+import java.util.concurrent.Callable;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Default task scheduler for {@link AbstractScheduledEventExecutor} that want to support task scheduling.
+ * Uses {@link DefaultPriorityQueue} to schdeule tasks.
+ */
+public final class DefaultTaskScheduler extends AbstractTaskScheduler {
+
+    private static final Comparator<RunnableScheduledFutureNode<?>> SCHEDULED_FUTURE_TASK_COMPARATOR =
+            Comparable::compareTo;
+    private static final RunnableScheduledFutureNode<?>[]
+            EMPTY_RUNNABLE_SCHEDULED_FUTURE_NODES = new RunnableScheduledFutureNode<?>[0];
```

Comment:
**Scalability Issue**: The DefaultTaskScheduler uses a PriorityQueue to manage scheduled tasks, which can become inefficient for a very high number of concurrent tasks due to its O(log n) complexity for insertion and removal operations. This could become a scalability bottleneck in systems with a high volume of scheduled tasks. <br> **Fix**: Implement or utilize a more efficient data structure for managing scheduled tasks, such as a time-wheel or a hierarchical timing wheel, which can offer better performance characteristics (O(1) complexity for insertion and removal) for the scheduling use case. <br> **Code Suggestion**: 
 ```
 private TaskScheduler taskScheduler;

 protected AbstractScheduledEventExecutor() {
 }

 protected AbstractScheduledEventExecutor(TaskSchedulerFactory taskSchedulerFactory) {
     this.taskSchedulerFactory = requireNonNull(taskSchedulerFactory, "taskSchedulerFactory");
     this.taskScheduler = taskSchedulerFactory.newTaskScheduler(this);
 }

 TaskScheduler taskScheduler() {
     return this.taskScheduler;
 }
 ```